src/richie/apps/search/filter_definitions/courses.py

```diff
@@ -202,6 +202,15 @@ def get_faceted_definitions(self, facets, *args, **kwargs):
         }
 
 
+class PersonsFilterDefinition(IndexableFilterDefinition):
+    """Persons are a special case of terms-based filter as they do not use MPTT paths."""
```

Comment:
Is it easy to make persons use IndexableFilterDefintion and the 2 others use a derived class that adds agg filtering?
Keep in mind that we will soon refactor nested indexables to not use path anymore...